### DISCUSSION TITLE
RecursionError when trying to call serialize_object() for a model which has m2m to DeviceType

### DISCUSSION THREAD

**Discussion by amyasnikov** [netbox-community/netbox#13785] [Help Wanted!]
### NetBox version

v3.6.1

### Python version

3.10

### Steps to Reproduce

1. Create a sample plugin. **models.py** will be enough, all other web-related stuff is not required
```
# models.py

from django.db import models
from netbox.models import NetBoxModel
from dcim.models import DeviceType


class SomeModel(NetBoxModel):
    device_types = models.ManyToManyField(DeviceType, related_name="+")
```
2. Create DeviceType object. All the properties doesn't matter
3. Create SomeModel object, bind device type from step 2 to it.
4. Try to call **serialize_object** for the object created at step 3

```
from dcim.models import DeviceType
from testplugin.models import SomeModel
from utilities.utils import serialize_object

device_type = DeviceType.objects.all()[0]

my_model = SomeModel.objects.create()
my_model.device_types.set([device_type])

serialize_object(my_model)  # RecursionError: maximum recursion depth exceeded
```

### Expected Behavior

**serialize_object(my_model)** doesn't lead to ~StackOverflow~  RecursionError

### Observed Behavior

RecursionError was raised.
As you know serialize_object() is called during form.save(). So this issue isn't just "theoretical"

**Comment by amyasnikov** [comment]
@jeremystretch why did you convert this issue into a discussion? I insist that this is a bug and it prevents plugin developers from using very basic functionality like ManyToMany links to DeviceType.

**Reply by jeremystretch** [reply]
We don't accept bug reports concerning custom code beyond what is [officially supported](https://docs.netbox.dev/en/stable/plugins/development/) for plugins. Additionally, it appears likely that your issue has already been captured under #13757.

**Comment by amyasnikov** [comment]
@jeremystretch if you need it I can show you more sophisticated example of this error when plugin also contains views.py, forms.py, urls.py, etc. This recursion error happens when you're trying to create a model instance with m2m to DeviceType (the error happens when backend receives POST data from the form)

**Comment by amyasnikov** [comment]
> We don't accept bug reports concerning custom code beyond what is [officially supported](https://docs.netbox.dev/en/stable/plugins/development/) for plugins. Additionally, it appears likely that your issue has already been captured under #13757.

This seems a bit strange to me that just a m2m relationship to core NetBox model is not supported, but it's nice (at least for me) that core NetBox also has this m2m relationship to DeviceType and so this recursion problem can't just be ignored
